### ISSUE TITLE
843. Guess the Word - Test Case Error

### ISSUE THREAD

**Issue by ChineduOkoronkwo** [LeetCode-Feedback/LeetCode-Feedback#688]
#### AlfredChinedu
AlfredChinedu

#### Category of the bug
- [X] Question
- [ ] Solution
- [ ] Language

#### Description of the bug

Based on the description of this problem, the test case below should return 2 when I call master.guess("aishiv"). It returns 0 instead, leading to errors in my solution.
"hbaczn"
["gaxckt","trlccr","jxwhkz","ycbfps","peayuf","yiejjw","ldzccp","nqsjoa","qrjasy","pcldos","acrtag","buyeia","ubmtpj","drtclz","zqderp","snywek","caoztp","ibpghw","evtkhl","bhpfla","ymqhxk","qkvipb","tvmued","rvbass","axeasm","qolsjg","roswcb","vdjgxx","bugbyv","zipjpc","tamszl","osdifo","dvxlxm","iwmyfb","wmnwhe","hslnop","nkrfwn","puvgve","rqsqpq","jwoswl","tittgf","evqsqe","aishiv","pmwovj","sorbte","hbaczn","coifed","hrctvp","vkytbw","dizcxz","arabol","uywurk","ppywdo","resfls","tmoliy","etriev","oanvlx","wcsnzy","loufkw","onnwcy","novblw","mtxgwe","rgrdbt","ckolob","kxnflb","phonmg","egcdab","cykndr","lkzobv","ifwmwp","jqmbib","mypnvf","lnrgnj","clijwa","kiioqr","syzebr","rqsmhg","sczjmz","hsdjfp","mjcgvm","ajotcx","olgnfv","mjyjxj","wzgbmg","lpcnbj","yjjlwn","blrogv","bdplzs","oxblph","twejel","rupapy","euwrrz","apiqzu","ydcroj","ldvzgq","zailgu","xgqpsr","wxdyho","alrplq","brklfk"]
10

#### Code you used for Submit/Run operation

```
class Solution {
    public class WordChar : IComparable<WordChar> {
        public char Key;
        public int Pop;
        public int QId;
        
        public WordChar(char key, int pop, int qid) {
            Key = key;
            Pop = pop;
            QId = qid;
        }
        
        public int CompareTo(WordChar b) {
            return this.Pop == b.Pop ? this.QId - b.QId : this.Pop - b.Pop;
        }
    }
    
    public void FindSecretWord(string[] wordlist, Master master) {
        var charToWord = new Dictionary<char, Dictionary<string, int>>();
        
        foreach(var word in wordlist) {
            foreach(var key in word) {
                if (!charToWord.ContainsKey(key)) 
                    charToWord.Add(key, new Dictionary<string, int>());
                
                if (!charToWord[key].ContainsKey(word))
                    charToWord[key].Add(word, 0);
                
                charToWord[key][word]++;
            }
        }
        
        var sortedSet = new SortedSet<WordChar>();
        foreach (var kvp in charToWord) {
            int qid = sortedSet.Count;
            int pop = kvp.Value.Count;
            sortedSet.Add(new WordChar(kvp.Key, pop, qid));
        }
        
        var dict = new HashSet<string>(wordlist);
        int count = 10;
        while (sortedSet.Count > 0 && dict.Count > 0 && count > 0) {
            
            int guess = 0;
            if (count >= dict.Count) {
                foreach(var wrd in dict) {
                    guess = master.guess(wrd);
                }
                break;
            }
            
            var wordChar = sortedSet.Max;
            var str = new string(wordChar.Key, 6);
            guess = master.guess(str);
            
            var charWords = charToWord[wordChar.Key];
            var rmv = new List<string>();
            foreach(var wrd in dict) {
                if (!charWords.ContainsKey(wrd) || charWords[wrd] != guess)
                    rmv.Add(wrd);
            }
            
            foreach(var wrd in rmv) dict.Remove(wrd);
            
            sortedSet.Remove(wordChar);
            count--;
        }
    }
}
```

/**
 * // This is the Master's API interface.
 * // You should not implement it, or speculate about its implementation
 * class Master {
 *     public int Guess(string word);
 * }
 */

#### Language used for code

C#

#### Expected behavior

I expected that at the point I called master.guess("aishiv") it should return 2. Instead it returned 0.

#### Additional context

The master.guess calls made by my solution and the returned values are in the sequence below.
"arabol", 1
"ppywdo", 0
"aishiv", 0

**Comment by lc-ivan** [comment]
Hi @ChineduOkoronkwo,
The code and test case you have provided does not return the value 0 or 2 when ran. Instead, it returns a "Wrong Answer" error. It seems that your submission may be missing certain conditions that are imperative to solve the problem.

**Comment by ChineduOkoronkwo** [comment]
Hi @Ic-ivan,

I am surprised and really disappointed that you closed this case without going through it. If you did go through the bug I logged you would have realised the statement "I expected that at the point I called master.guess("aishiv") it should return 2. Instead it returned 0."

Yes, the final result of executing my code is wrong answer. The question is why is it returning wrong answer with this test case? The bug is that executing master.guess("aishiv") return 0 when it should return 2 with the test case in question.
